app/Http/Controllers/EventController.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Storage;
+use App\Models\Event;
+use App\Models\Level;
+
+class EventController extends Controller
+{
+    const LOCAL_STORAGE_FOLDER = 'public/images/';
+    private $event;
+    private $level;
+
+    
+
+    public function __construct(Event $event, Level $level){
+        $this->event = $event;
+        $this->level = $level;
+    }
+
+    public function show(){
+        $all_events = $this->event->all();
+
+        return view('users.event')
+        ->with('all_events' , $all_events);
+    }
+
+    public function showDetail($id){
```

Comment:
here you can just say `show(Event $event)` and you can now delete `$event = $this->event->findOrFail($id);`

Reply:
once updated, please update your `web.php`

Reply:
once every comment is implemented you can remove the entire

```
private $event;
private $level;
```

```
public function __construct(Event $event, Level $level){
        $this->event = $event;
        $this->level = $level;
}
```